app/src/features/surveys/observations/sampling-sites/list/import-observations/ImportObservationsButton.tsx

```diff
@@ -0,0 +1,146 @@
+import { mdiImport } from '@mdi/js';
+import Icon from '@mdi/react';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import FileUploadDialog from 'components/dialog/FileUploadDialog';
+import { UploadFileStatus } from 'components/file-upload/FileUploadItem';
+import { ObservationsTableI18N } from 'constants/i18n';
+import { DialogContext } from 'contexts/dialogContext';
+import { SurveyContext } from 'contexts/surveyContext';
+import { useBiohubApi } from 'hooks/useBioHubApi';
+import { useContext, useState } from 'react';
+
+export interface IImportObservationsButtonProps {
+  /**
+   * If true, the button will be disabled.
+   *
+   * @type {boolean}
+   * @memberof IImportObservationsButtonProps
+   */
+  disabled?: boolean;
+  /**
+   * Callback fired when the import process is started.
+   *
+   * @memberof IImportObservationsButtonProps
+   */
+  onStart?: () => void;
+  /**
+   * Callback fired when the import process is successful.
+   *
+   * @memberof IImportObservationsButtonProps
+   */
+  onSuccess?: () => void;
+  /**
+   * Callback fired when the import process encounters an error.
+   *
+   * @memberof IImportObservationsButtonProps
+   */
+  onError?: () => void;
+  /**
+   * Callback fired when the import process is complete (success or error).
+   *
+   * @memberof IImportObservationsButtonProps
+   */
+  onFinish?: () => void;
+  /**
+   * Options to pass to the process csv submission endpoint.
+   *
+   * @type {{
+   *     surveySamplePeriodId?: number;
+   *   }}
+   * @memberof IImportObservationsButtonProps
+   */
+  processOptions?: {
+    /**
+     * The optional survey sample period id to associate to all parsed observation records.
+     *
+     * @type {number}
+     */
+    surveySamplePeriodId?: number;
+  };
+}
+
+export const ImportObservationsButton = (props: IImportObservationsButtonProps) => {
+  const { disabled, onStart, onSuccess, onError, onFinish, processOptions } = props;
+
+  const biohubApi = useBiohubApi();
+
+  const surveyContext = useContext(SurveyContext);
+  const { projectId, surveyId } = surveyContext;
+
+  const dialogContext = useContext(DialogContext);
+
+  const [open, setOpen] = useState<boolean>(false);
+
+  /**
+   * Callback fired when the user attempts to import observations.
+   *
+   * @param {File} file
+   * @return {*}
+   */
+  const handleImportObservations = async (file: File) => {
```

Comment:
Nit: A try catch here might make this slightly more readable.

Reply:
Updated

Reply:
Had 2 nearly identical components doing the same thing, so I've merged them into one file in a new folder.